Dart/thirdPartySrc/analysisServer/org/dartlang/analysis/server/protocol/Outline.java

```diff
@@ -96,9 +103,11 @@ public static Outline fromJson(Outline parent, JsonObject outlineObject) {
     Element element = Element.fromJson(elementObject);
     int offset = outlineObject.get("offset").getAsInt();
     int length = outlineObject.get("length").getAsInt();
+    int codeOffset = outlineObject.get("codeOffset").getAsInt();
+    int codeLength = outlineObject.get("codeLength").getAsInt();
```

Comment:
Aren't `codeOffset` and `codeLenght` optional? I had to change these 2 lines to
```
    int codeOffset = outlineObject.get("codeOffset") == null ? offset : outlineObject.get("codeOffset").getAsInt();
    int codeLength = outlineObject.get("codeLength") == null ? length : outlineObject.get("codeLength").getAsInt();
```
in order not to crash with older SDKs.